From 5cf18a5fe50b5d58417aba8d64def90788a7eab8 Mon Sep 17 00:00:00 2001
From: Matthias Schoepfer <m.schoepfer@rethinkrobotics.com>
Date: Mon, 29 Jan 2024 17:09:50 +0100
Subject: [PATCH] remove insane string replace

Removing string replace where totally fine #!/usr/bin/env python was
replaced with the native python executable.

Signed-off-by: Matthias Schoepfer <m.schoepfer@rethinkrobotics.com>
---
 CMakeModules/FindPython.cmake | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/CMakeModules/FindPython.cmake b/CMakeModules/FindPython.cmake
index 6f99625..7c111b8 100644
--- a/CMakeModules/FindPython.cmake
+++ b/CMakeModules/FindPython.cmake
@@ -57,8 +57,6 @@ endif(NOT PYTHON_EXEC)
 get_filename_component(PYTHON_EXEC_ "${PYTHON_EXEC}" REALPATH)
 set(PYTHON_EXEC "${PYTHON_EXEC_}" CACHE FILEPATH "Path to Python interpreter")
 
-string(REGEX REPLACE "/bin/python.*" "" PYTHON_PREFIX "${PYTHON_EXEC_}")
-string(REGEX REPLACE "/bin/python.*" "" PYTHON_PREFIX2 "${PYTHON_EXEC}")
 
 if(PYTHON_EXEC)
     execute_process(COMMAND "${PYTHON_EXEC}" "-c"
@@ -206,7 +204,6 @@ macro(install_python)
         cmake_parse_arguments(install_python "" "DESTINATION;COMPONENT;RENAME" "PROGRAMS" "${ARGN}")
         foreach(script ${install_python_PROGRAMS})
             file(READ ${script} _contents)
-            string(REPLACE "#!/usr/bin/env python" "#!${PYTHON_EXEC}" _fixed "${_contents}")
             get_filename_component(_realscript "${script}" NAME)
             file(WRITE "${PROJECT_BINARY_DIR}/${install_python_DESTINATION}/${_realscript}" "${_fixed}")
             install(PROGRAMS "${PROJECT_BINARY_DIR}/${install_python_DESTINATION}/${_realscript}"
-- 
2.34.1

